Feature/Playground/Playground/Models/Playground.swift

```diff
@@ -0,0 +1,105 @@
+import UIKit
+
+/// `UIDocument` subclass used to manage `.ocelotplayground` files.
+public final class Playground: UIDocument {
+	/// Contents of the playgrounds `contents.graphql` file.
+	private (set) var contents = ""
+
+	/// `Settings` stored inside the playgrounds `settings.json` file.
+	private (set) var settings = Settings(endpoint: nil)
+
+	private lazy var jsonEncoder = JSONEncoder()
+	private lazy var jsonDecoder = JSONDecoder()
+
+	private lazy var packageFileWrapper = FileWrapper(directoryWithFileWrappers: [:])
+
+	/// Tells the `Playground` to store the given contents on the next save.
+	///
+	/// - Warning: This will not save the updated Playground to disk.
+	/// - Parameter updatedContents: The updated contents.
+	public func updateContents(_ updatedContents: String) {
+		contents = updatedContents
+
+		if let contentsFileWrapper = packageFileWrapper.fileWrappers?[FileName.contents] {
+			packageFileWrapper.removeFileWrapper(contentsFileWrapper)
+		}
+	}
+
+	/// Tells the `Playground` to store the given `Settings` on the next save.
+	///
+	/// - Warning: This will not save the updated Playground to disk.
+	/// - Parameter updatedSettings: The updated `Settings`.
+	public func updateSettings(_ updatedSettings: Settings) {
+		settings = updatedSettings
+
+		if let settingsFileWrapper = packageFileWrapper.fileWrappers?[FileName.settings] {
+			packageFileWrapper.removeFileWrapper(settingsFileWrapper)
+		}
+	}
+
+	public override func contents(forType typeName: String) throws -> Any {
+		if packageFileWrapper.fileWrappers?[FileName.contents] == nil,
+			let contentsData = contents.data(using: .utf8
+		) {
+			let contentsFileWrapper = FileWrapper(regularFileWithContents: contentsData)
+			contentsFileWrapper.preferredFilename = FileName.contents
+			packageFileWrapper.addFileWrapper(contentsFileWrapper)
+		}
+
+		if let settingsData = try? jsonEncoder.encode(settings) {
```

Comment:
Here we should check whether a FileWrapper for settings already exists to prevent unnecessary encoding. Instead of defining update methods we could also put this logic inside property observers of the respective properties (`contents` & `settings`).

Reply:
Agreed. ✅ We should add it as property observers instead. That way we can leave the setters for both `contents` and `settings` `public` which will be more transparent to use.